ngraph_bridge/ngraph_freshness_tracker.cc

```diff
@@ -69,10 +69,10 @@ void NGraphFreshnessTracker::RemoveTensor(const void* base_pointer) {
 }
 
 void NGraphFreshnessTracker::RemoveUser(
-    std::shared_ptr<ngraph::runtime::Executable> user) {
+    const std::shared_ptr<ngraph::runtime::Executable>& user) {
   mutex_lock l(mu_);
-  for (auto kv : freshness_map_) {
-    kv.second.erase(user);
+  for (auto& kv : freshness_map_) {
```

Comment:
change `kv` to `& kv` is the fix for this problem. The rest of change is just for performance and simplicity.
In short, if using `kv`, we just erase user from the copy of `ngraph::Executable` set. The original freshness_map is untouched. Period.

Reply:
Good Catch! :)

Reply:
Good catch! Thanks for debugging this :)